### PR TITLE
Upgrate to cosmos-sdk v0.50 and ibc-go v8

### LINKED ISSUE #161: Upgrade to cosmos-sdk v0.50
Upgrade fee-abstraction to support cosmos-sdk v0.50.x

### PR DESCRIPTION
**Pull request by tnv1**
Closed #161.

### CODE REVIEW

**Comment by nghuyenthevinh2000** [comment]
according to repo structure, release/v8.0.x will contain sdk v50 upgrade & ibc-go v8

File: app/app.go [hunk 1]
```diff
@@ -327,106 +355,112 @@
...
 	)
 	app.StakingKeeper = stakingkeeper.NewKeeper(
 		appCodec,
-		keys[stakingtypes.StoreKey],
+		runtime.NewKVStoreService(keys[stakingtypes.StoreKey]),
 		app.AccountKeeper,
 		app.BankKeeper,
-		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		authority,
+		authcodec.NewBech32Codec(sdk.Bech32PrefixValAddr),
```
[the remark this answers is not in the document]
Comment (tnv1) [reply app/app.go:382]: no need warning

File: app/app.go [hunk 2]
```diff
@@ -983,10 +957,43 @@
...
 
 	return paramsKeeper
 }
+
+// GetStoreKeys returns all the stored store keys.
+func (app *FeeApp) GetStoreKeys() []storetypes.StoreKey {
+	keys := make([]storetypes.StoreKey, 0, len(app.keys))
+	for _, key := range app.keys {
+		keys = append(keys, key)
+	}
```
[the remark this answers is not in the document]
Comment (tnv1) [reply app/app.go:937]: no need

File: app/app.go [hunk 3]
```diff
@@ -983,10 +957,43 @@
...
+func (app *FeeApp) AutoCliOpts() autocli.AppOptions {
+	modules := make(map[string]appmodule.AppModule, 0)
+	for _, m := range app.mm.Modules {
+		if moduleWithName, ok := m.(module.HasName); ok {
+			moduleName := moduleWithName.Name()
+			if appModule, ok := moduleWithName.(appmodule.AppModule); ok {
+				modules[moduleName] = appModule
+			}
+		}
+	}
```
[the remark this answers is not in the document]
Comment (tnv1) [reply app/app.go:954]: no need

File: app/app.go [hunk 4]
```diff
@@ -983,10 +923,69 @@
...
+
+	return dupMaccPerms
+}
+
+// BlockedAddresses returns all the app's blocked account addresses.
+func BlockedAddresses() map[string]bool {
+	modAccAddrs := make(map[string]bool)
+	for acc := range GetMaccPerms() {
+		modAccAddrs[authtypes.NewModuleAddress(acc).String()] = true
+	}
```
[the remark this answers is not in the document]
Comment (tnv1) [reply app/app.go:982]: no need

File: app/app.go [hunk 5]
```diff
@@ -327,106 +355,112 @@
...
 	app.StakingKeeper = stakingkeeper.NewKeeper(
 		appCodec,
-		keys[stakingtypes.StoreKey],
+		runtime.NewKVStoreService(keys[stakingtypes.StoreKey]),
 		app.AccountKeeper,
 		app.BankKeeper,
-		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		authority,
+		authcodec.NewBech32Codec(sdk.Bech32PrefixValAddr),
+		authcodec.NewBech32Codec(sdk.Bech32PrefixConsAddr),
```
[the remark this answers is not in the document]
Comment (tnv1) [reply app/app.go:383]: no need

File: app/app.go [hunk 6]
```diff
@@ -327,48 +340,53 @@
...
 
 	// add keepers
 	app.AccountKeeper = authkeeper.NewAccountKeeper(
 		appCodec,
-		keys[authtypes.StoreKey],
+		runtime.NewKVStoreService(keys[authtypes.StoreKey]),
 		authtypes.ProtoBaseAccount,
 		maccPerms,
-		sdk.Bech32PrefixAccAddr,
+		authcodec.NewBech32Codec(sdk.Bech32MainPrefix),
```
[the remark this answers is not in the document]
Comment (tnv1) [reply app/app.go:353]: no need

File: app/app.go [hunk 7]
```diff
@@ -983,10 +923,69 @@
...
+}
+
+// GetMaccPerms returns a copy of the module account permissions
+//
+// NOTE: This is solely to be used for testing purposes.
+func GetMaccPerms() map[string][]string {
+	dupMaccPerms := make(map[string][]string)
+	for k, v := range maccPerms {
+		dupMaccPerms[k] = v
+	}
```
[the remark this answers is not in the document]
Comment (tnv1) [reply app/app.go:972]: no need

**Review by minhngoc274 — APPROVED** [review]

**Pull request merged**